parlai/tasks/qangaroo/agents.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.teachers import DialogTeacher, FixedDialogTeacher
+from .build import build
+
+import json
+import os
+
+class IndexTeacher(FixedDialogTeacher):
+    def __init__(self, opt, shared=None):
+        datapath = self._path(opt)
+        super().__init__(opt, shared)
+        self.data = self._setup_data(datapath)
+        self.id = 'qangaroo'
+        self.reset()
+
+    def _path(self, opt):
+        # Build the data if it doesn't exist.
+        build(opt)
+        dt = opt['datatype'].split(':')
+        datatype = 'dev' if dt[0] == 'valid' else 'train'
```

Comment:
let's reverse this so that 'test' maps to 'dev' rather than 'train': 
`datatype = 'train' if dt[0] == 'train' else 'dev'